utils/logalloc.cc

```diff
@@ -1319,7 +1340,18 @@ segment_pool::segment_pool(tracker::impl& tracker)
 {
 }
 
-void segment_pool::prime(size_t available_memory, size_t min_free_memory) {
+void segment_pool::prime(size_t available_memory, size_t min_free_memory, std::optional<segment_pool_backend> select_backend) {
+#ifndef SEASTAR_DEFAULT_ALLOCATOR
+    if (select_backend) {
+        if (_segments_in_use) {
+            throw std::runtime_error("cannot change segment store backend if segments are in use");
+        }
+        _store.select_backend(*select_backend, available_memory);
+        _segments = std::vector<segment_descriptor>(max_segments());
+        _lsa_owned_segments_bitmap = utils::dynamic_bitset(max_segments());
+        _lsa_free_segments_bitmap = utils::dynamic_bitset(max_segments());
+    }
+#endif
     auto old_emergency_reserve = std::exchange(_emergency_reserve_max, std::numeric_limits<size_t>::max());
```

Comment:
Should we even prime the pool for tools? That's intended for low-latency operations, and is trying to segregate segments into top of memory and non-lsa into the bottom. Tools don't share memory space. We should just return of we're using the standard allocator.

Reply:
Right, we can allocate the segments later on-demand. If we do that, your suggestion above to prefer the freelist might indeed help. Will change in v2.

Reply:
I will extract this backend selection into a method of its own, and won't call `prime_segment_pool() ` for tools.

Reply:
> If we do that, your suggestion above to prefer the freelist might indeed help. Will change in v2.

It won't (though I think the code will look more natural). The only thing that will free segments is cache reclaim, and that is triggered by running out of segments.

Reply:
btw, please test tools with a segment pool holding just one segment and something that needs lots of cache (what?!) to see that reclaim actually works and wasn't broken somehow.

Reply:
Ok, I will dedicated unit tests for this. Previously (when this was replacing the debug store), this was tested via the regular debug builds, but this is no longer the case.